### PR TITLE
Use formik for login and sign up forms

### LINKED ISSUE #161: Try using formik for form & validation
See https://github.com/jaredpalmer/formik

### PR DESCRIPTION
**Pull request by phyrog**
Closes #161.

I had to remove some tests for the `SignInForm` and `SignUpForm` as I couldn't find any way on how to wait for validation/submission to finish to check the changes in output. This happens all asynchronously and I don't have access to the Promises, so I'm out of ideas here.

### CODE REVIEW

**Comment by phyrog** [comment]
I'm currently reworking this to be better testable. Not ready for review yet.

File: src/components/SignUpForm/validation.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      }
+    });
+};
+
+export const setServerErrors = (setErrors, setSubmitting, onError) => err => {
+  let errors = {};
+  let matches;
+  let regex = /GraphQL error:\s*((\S+) .+)\s*$/gm;
+
+  while ((matches = regex.exec(err.message))) {
```
[the remark this answers is not in the document]
Comment (phyrog) [reply src/components/SignUpForm/validation.js]: This is wanted in this case.

[the remark this answers is not in the document]
Comment (eugenk) [reply src/components/SignUpForm/validation.js]: Then let's use
```javascript
/* eslint-disable no-cond-assign */
while ((matches = regex.exec(err.message))) {
/* eslint-enable no-cond-assign */
 ```

File: src/components/SignUpForm/index.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  captcha: Captcha,
+  errors,
+  onBlur,
+  onChange,
+  onSubmit,
+  submitDisabled,
+  touched,
+  values
+}) => (
+  <Form onSubmit={!submitDisabled ? onSubmit : undefined}>
```
Comment (eugenk) [inline src/components/SignUpForm/index.js]: Isn't it more readable with one less negation? ;)
```javascript
submitDisabled ? undefined : onSubmit
```
or even:
```javascript
submitEnabled ? onSubmit : undefined
```

**Comment by phyrog** [comment]
I rebased on `master` and added commit 5cb14e7 to address the review.

**Pull request merged**